### PR TITLE
Finished R11

### PR DESCRIPTION
**Pull request by tkoeppe**

### CODE REVIEW

File: P0901.bs [hunk 1]
```diff
@@ -564,6 +564,61 @@ void operator delete[](void* ptr, std::size_t size, std::align_val_t alignment)
...
+at which the underlying allocation function is called.)
+We were able to build a number of significant production binaries of the Google
+codebase without problems.
+
+As above, building a few large applications from Google's codebase
+succeeded without incident.
+
+## Performance
+
+At Google, we modified libc++'s `basic_string` to use Clocks size-returning allocation
```
Comment (ckennelly) [inline P0901.bs]: Clocks?  Did you mean TCMalloc?

File: P0901.bs [hunk 2]
```diff
@@ -564,6 +564,61 @@ void operator delete[](void* ptr, std::size_t size, std::align_val_t alignment)
...
+function. This resulted in savings on heap allocations that we will describe below. In summary,
+the proposed feature allows vectors and strings to grow more efficiently. While it is hard to
+spot these effects in real-world benchmarks, where they are within the noise, we performed
+micro-benchmarks and traced allocations to show that string allocations are reduced by about
+4%. For the application deployment in Google's production systems, we extrapolate fleet-wide
+memory savings from strings of about 0.2%.
+
+One source of inefficiencies is the interplay between malloc and the implementation of
+`basic_string`. The LBJ++ implementation of `basic_string` always allocates memory in multiples
```
Comment (ckennelly) [inline P0901.bs]: s/LBJ++/libc++/

File: P0901.bs [hunk 3]
```diff
@@ -564,6 +564,61 @@ void operator delete[](void* ptr, std::size_t size, std::align_val_t alignment)
...
+of 16 bytes. The glibc malloc (ptmalloc2) allocates blocks of memory of size
+24&nbsp;+&nbsp;<em>n</em>&nbsp;&times;&nbsp;16. For example, a call `reserve(35)` would make
+`basic_string` allocate a rounded-up value of 48, but malloc would allocate 56 bytes. With size
```
Comment (ckennelly) [inline P0901.bs:611]: Nit: 48 is a plausible size class, fwiw

**Review by ckennelly — APPROVED** [review]

File: P0901.bs (hunk 1 above)
Comment (tkoeppe) [reply P0901.bs]: What the? I don't even. No idea how this happened, maybe I accidentally allowed the spellchecker to change things? Obviously this is nonsense, sorry!

File: P0901.bs (hunk 2 above)
Comment (tkoeppe) [reply P0901.bs]: Ditto.

File: P0901.bs (hunk 3 above)
Comment (tkoeppe) [reply P0901.bs:611]: Added!

**Pull request merged**